tests/core/v5_1/test_client.py

```diff
@@ -237,6 +240,138 @@ async def _send_response():
     assert len(checked_bucket_indexes) > 4
 
 
+@pytest.mark.trio
+async def test_client_request_response_stream_find_nodes_found_nodes(
+    alice, bob, alice_client, bob_client
+):
+    table = KademliaRoutingTable(bob.node_id, 256)
+    for i in range(1000):
```

Comment:
I think this test is probably overkill.  Here's what I think you want to make sure you are testing.

1. `ValidationError` case for returning ENR records with the wrong distance.
2. `ValidationError` case for `distance == 0` case
3. `ValidationError` case for `total == 0`
4. `ValidationError` for multiple responses with different `total` fields
5. That you can consume the full stream the full set of responses/ENR records
6. That you can consume only a few records out of a stream of response/ENR records and exit the context early
7. That we get a `TimeoutError` when the response messages don't show up

For these I think you can just create a few ENR records to populate the routing table and then request those specific distances.